### DISCUSSION TITLE
Comparing the concepts from CMSIS-Pack and xPack

### DISCUSSION THREAD

**Discussion by fred-r** [Open-CMSIS-Pack/Open-CMSIS-Pack-Spec#50] [Ideas]
The intention is to compare the concepts of CMSIS-pack and xPack and understand the salient differences.
(https://xpack.github.io/)

Starting from this comment:
https://github.com/Open-CMSIS-Pack/Open-CMSIS-Pack/discussions/48#discussioncomment-1687925

**Comment by fred-r** [comment]
https://open-cmsis-pack.github.io/Open-CMSIS-Pack/main/html/index.html

==> CMSIS-pack "alone" aims mainly at delivering software.

But the Open-CMSIS-pack project targets more:
https://github.com/Open-CMSIS-Pack/Open-CMSIS-Pack/issues/6

Probably @jkrech or @ReinhardKeil can explain much better than me the scope of the project.

@ilg-ul : is xPack having the same objectives ?

**Comment by 0Grit** [comment]
https://xpack.github.io/intro/#but-what-are-xpacks

**Comment by ilg-ul** [comment]
> is xPack having the same objectives

The xPack project (which now has the full name of _The xPack Build Framework_) aims to provide a cross-platform solution to manage dependencies and builds, a solution simple enough to be used by end-users in their projects.

My personal interest is in C++ embedded projects, but the design is currently language neutral.

The basic design aimed to be an extension of the npm packages, as the most successful package format, with billions of downloads.

The first layer handles satisfying dependencies to source and binary packages, so that on a freshly installed host system, running `xpm install` will bring all necessary dependencies, allowing to directly run builds, tests, etc.

Dependencies can be defined for the entire project, or for each build configuration, in other words different build configurations can be used for different architectures/platforms.

You can take a look at some functional examples in the VS Code xPack C/C++ Build Extension page:

- https://xpack.github.io/vscode/quick-start/

The project is still work in progress, so nothing is cast in stone.
